src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,40 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Prints the given `str` and reads a [`String`] from
+/// [standard input](Stdin). The trailing newline is stripped.
+/// Gives an error on EOF (Ctrl-D).
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(input)]
+/// use std::io::input;
+///
+/// let name = input("Enter name: ").expect("input failed");
+///
+/// println!("Your name is {}!", name);
+/// ```
+#[unstable(
+    feature = "input",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn input(prompt: &str) -> Result<String> {
+    stdout().write_all(prompt.as_bytes())?;
+    stdout().flush()?;
```

Comment:
Currently this locks stdout twice.
```suggestion
    let mut stdout = stdout().lock();
    stdout.write_all(prompt.as_bytes())?;
    stdout.flush()?;
```